src/errors/fieldErrors.ts

```diff
@@ -22,3 +23,17 @@ export class FieldTypeInconsistencyError extends PushApiClientBaseError {
       Make sure to review and fix your document metadata before pushing`;
   }
 }
+
+export class InvalidPermanentId extends PushApiClientBaseError {
+  public name = 'Invalid permanentId field Error';
+  public constructor(field: FieldModel) {
+    super(`
+    The detected permanentid field in the index is incorrectly configured.
```

Comment:
(just a bit easier to read:)
```suggestion
    The permanentid field detected in the index is not correctly configured.
```

Reply:
Thanks!